### PR TITLE
Updated data handler

### PR DESCRIPTION
**Pull request by marianfaryna**

### CODE REVIEW

File: rest-client/src/main/scala/com/dota/rest/client/DataHandler.scala [hunk 1]
```diff
@@ -1,41 +1,70 @@
...
-    Thread.sleep(5000)
-    val document: Document = Document("_id" -> id, "value" -> value)
-    val insertObservable: Observable[Completed] = collection.insertOne(document)
-    Thread.sleep(5000)
-    insertObservable.subscribe(new Observer[Completed] {
-      override def onNext(result: Completed): Unit = println(s"onNext insertObservable: $result")
-      override def onError(e: Throwable): Unit = println(s"onError insertObservable: $e")
-      override def onComplete(): Unit = println("insertObservable onComplete")
-    })
+  val client: MongoClient = MongoClient("mongodb://localhost:27017") //the actual host name of parent machine
```
Comment (oshtabura) [inline rest-client/src/main/scala/com/dota/rest/client/DataHandler.scala]: Maybe lets use incoming param for this ?
We can use default one for port for example

File: rest-client/src/main/scala/com/dota/rest/client/DataHandler.scala [hunk 2]
```diff
@@ -1,41 +1,70 @@
...
+
+  val database = client.getDatabase("local")
+  val collection: MongoCollection[Document] = database.getCollection("test_collection")
```
Comment (oshtabura) [inline rest-client/src/main/scala/com/dota/rest/client/DataHandler.scala]: Move constants to some class

File: rest-client/src/main/scala/com/dota/rest/client/DataHandler.scala [hunk 3]
```diff
@@ -1,41 +1,70 @@
...
+
+  def prepareDocument(matchToPersist: Match): Document = {
+    Document("_id" -> matchToPersist.matchId,
+      "radiantWin" -> matchToPersist.radiantWin,
```
Comment (oshtabura) [inline rest-client/src/main/scala/com/dota/rest/client/DataHandler.scala]: Same as previous comment

File: rest-client/src/main/scala/com/dota/rest/client/DataHandler.scala [hunk 4]
```diff
@@ -1,41 +1,70 @@
...
+      "direScore" -> matchToPersist.direScore,
+      "players" -> preparePlayersDocument(matchToPersist.players))
+  }
+
+  def preparePlayersDocument(players: List[Player]): List[Document] = {
+    var playersList = List[Document]()
+    for (player <- players) {
+      player.defineSide()
+      playersList = playersList :+ Document("accountId" -> player.accountId,
+        "heroId" -> player.heroId,
```
Comment (oshtabura) [inline rest-client/src/main/scala/com/dota/rest/client/DataHandler.scala]: Same as previous comment

File: rest-client/src/main/scala/com/dota/rest/client/DataHandler.scala [hunk 5]
```diff
@@ -1,41 +1,70 @@
...
+        "isDire" -> player.isDire
+      )
+    }
+    playersList
+  }
+
+  def put(matchToPersist: Match): Unit = {
+    val findObservable = collection.find(equal("_id", matchToPersist.matchId)).first()
+
+    val foundDocument = Await.result(findObservable.toFuture(), Duration(10, TimeUnit.SECONDS))
```
Comment (oshtabura) [inline rest-client/src/main/scala/com/dota/rest/client/DataHandler.scala]: Maybe lets move 10 second to some configuration, so we could handle all timeout in a single place ?

File: rest-client/src/main/scala/com/dota/rest/client/RestClient.scala [hunk 6]
```diff
@@ -15,27 +15,30 @@ class RestClient {
...
-  val MATCH_DETAILS_URL = DOTA_BASIC_REST_URL +"GetMatchDetails/V001/?match_id=%s&key=%s"
-  val ITEMS_URL = DOTA_ECON_BASIC_REST_URL + "GetGameItems/V001/?key=%s"
-  val HEROES_URL = DOTA_ECON_BASIC_REST_URL + "GetHeroes/V001/?key=%s"
-  val ABILITIES_URL = "https://github.com/kronusme/dota2-api/blob/master/data/abilities.json"
+  var LatestMatchId :String = _ //start_at_match_id=${match_id}
+
+  var steamKey :String = _
+//todo constant StartAtMatchId
+  val StartAtMatchId = "?start_at_match_id="
+  val DotaBasicRestUrl = "https://api.steampowered.com/IDOTA2Match_570/"
```
Comment (oshtabura) [inline rest-client/src/main/scala/com/dota/rest/client/RestClient.scala]: Maybe lets move all constants to single place

**Review by oshtabura — COMMENTED** [review]
Looks good, just few minor things

File: rest-client/src/main/scala/com/dota/rest/client/DataHandler.scala (hunk 1 above)
Comment (marianfaryna) [reply rest-client/src/main/scala/com/dota/rest/client/DataHandler.scala]: like command-line argument or some property file in resources?

File: rest-client/src/main/scala/com/dota/rest/client/DataHandler.scala (hunk 5 above)
Comment (marianfaryna) [reply rest-client/src/main/scala/com/dota/rest/client/DataHandler.scala]: like command-line argument? or configuration file?

File: rest-client/src/main/scala/com/dota/rest/client/DataHandler.scala (hunk 1 above)
Comment (oshtabura) [reply rest-client/src/main/scala/com/dota/rest/client/DataHandler.scala]: I believe it's better as configuration file

File: rest-client/src/main/scala/com/dota/rest/client/DataHandler.scala (hunk 5 above)
Comment (oshtabura) [reply rest-client/src/main/scala/com/dota/rest/client/DataHandler.scala]: I believe it's better as configuration file

File: rest-client/src/main/scala/com/dota/rest/client/Constants.scala [hunk 7]
```diff
@@ -0,0 +1,17 @@
+package com.dota.rest.client
+
+/**
+  * Constants that are used in applicaiton that are not configurable
+  */
+object Constants {
```
Comment (oshtabura) [inline rest-client/src/main/scala/com/dota/rest/client/Constants.scala:6]: maybe rename to RestUrlConstants

**Review by oshtabura — APPROVED** [review]
Please merge

File: rest-client/src/main/scala/com/dota/rest/client/Constants.scala (hunk 7 above)
Comment (marianfaryna) [reply rest-client/src/main/scala/com/dota/rest/client/Constants.scala:6]: I thought all other constants that we will need, we be also there

**Pull request merged**